cmd: add zsh as a option for completion

tonyluj opened the pull request:
This adds zsh as a option for command completion, for zsh is also
popular and cobra natively supports zsh.

This will not change the default behavior of "completion" command.
If no or "bash" argument specified, it generates the completion for
bash.

Signed-off-by: Tony Lu <tonylu@linux.alibaba.com>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/9882)
<!-- Reviewable:end -->

tonyluj committed:
cmd: add zsh as a option for completion

This adds zsh as a option for command completion, for zsh is also
popular and cobra natively supports zsh.

This will not change the default behavior of "completion" command.
If no or "bash" argument specified, it generates the completion for
bash.

Signed-off-by: Tony Lu <tonylu@linux.alibaba.com>

tonyluj commented on cilium/cmd/root.go:
```diff
@@ -155,6 +167,16 @@ func runCompletion(out io.Writer, cmd *cobra.Command, args []string) error {
+	}
 
-	return cmd.Parent().GenBashCompletion(out)
+	return fmt.Errorf("Unexpected shell: %s.", args[0])
```
Just follow the error format in this function. Maybe we could change them all.

```go
func runCompletion(out io.Writer, cmd *cobra.Command, args []string) error {
	if len(args) > 1 {
		return fmt.Errorf("Too many arguments. Expected only the shell type.") // <--------
	}
	if _, err := out.Write([]byte(copyRightHeader)); err != nil {
		return err
	}
	if len(args) == 0 {
		return cmd.Parent().GenBashCompletion(out)
	}

	switch args[0] {
	case "bash":
		return cmd.Parent().GenBashCompletion(out)
	case "zsh":
		return cmd.Parent().GenZshCompletion(out)
	}

	return fmt.Errorf("Unexpected shell: %s.", args[0])
}
```

rolinh replied:
Yes, feel free to fix the other one too.
Note that the error is actually unused. Would you mind changing the call as follows so that errors are actually taken into account?
```go
RunE: func(cmd *cobra.Command, args []string) error {
    return runCompletion(out, cmd, args)
},
```

tonyluj replied:
Yep, I am going to change Run to RunE, and return the error msg to outer.

aanm reviewed (approved):
LGTM as long @Rolinh changes requested are taken into account

tonyluj committed:
cmd: expose completion error to outer

This replaces Command.Run() with Command.RunE(), which exposes the error
to the outer scope, then the rootCmd.Execute() could catch this error
and print the error message with exit code -1.

Signed-off-by: Tony Lu <tonylu@linux.alibaba.com>

aanm reviewed (changes requested):
@tonyluj please run `make -C Documentation cmdref` and re-submit your changes. Thank you!
